main/actions/src/main/scala/sbt/Tests.scala

```diff
@@ -273,7 +273,19 @@ object Tests {
   def discover(frameworks: Seq[Framework], analysis: CompileAnalysis, log: Logger): (Seq[TestDefinition], Set[String]) =
     discover(frameworks flatMap TestFramework.getFingerprints, allDefs(analysis), log)
 
-  def allDefs(analysis: CompileAnalysis) = analysis match { case analysis: Analysis => analysis.apis.internal.values.flatMap(_.api.definitions).toSeq }
+  def allDefs(analysis: CompileAnalysis) = analysis match {
```

Comment:
This method used to return top level classes and objects but now returns also nested and inherited. Is that change intentional?

Have considered using this definition?

```scala
def allDefs(analysis: CompileAnalysis): Seq[xsbti.api.ClassLike] = {
  def companionsApis(c: xsbti.api.Companions) = Seq(c.classApi, c.objectApi)
  analysis match {
    case analysis: Analysis =>
      analysis.apis.internal.values.flatMap(x => companionsApis(x.api)).toSeq

  }
}
```